lib/Cake/Console/Helper/ProgressShellHelper.php

```diff
@@ -115,8 +115,11 @@ public function draw() {
 		if ($pad > 0) {
 			$bar .= str_repeat(' ', $pad);
 		}
-		$percent = ($complete * 100) . '%';
-		$bar .= str_pad($percent, $numberLen, ' ', STR_PAD_LEFT);
-		$this->_consoleOutput->overwrite($bar, 0);
+		$percent = ($complete * 100);
+		$bar .= str_pad($percent . '%', $numberLen, ' ', STR_PAD_LEFT);
+		if ($percent % 10 === 0 && $percent !== $logged_percentage) {
```

Comment:
What is this check for @JINGJINGTANG ?
`$percent !== $logged_percentage`

Reply:
This is to make sure we are only logging the same progress (percentage) once to the log file. Since we are logging the progress to a log file, we can't have it overwritten the previous log. Thus, we are logging 0% ---> 10% ---> 20%...etc

Reply:
In the mcap parse, we would have this inside the for loop: `$logged_percentage = $progress->draw($logged_percentage);`